### PR TITLE
feat: Add support for gVisor per node pool

### PR DESCRIPTION
**Pull request by LukaszCzarnotaSabre**
Issue [https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/issues/240](url) was resolved but now all node pools needs to be with gVisor or non of them.

With this PR you are able to define `sandbox_enabled` variable per node pool, code is backward compatible.

### CODE REVIEW

**Comment by LukaszCzarnotaSabre** [comment]
@googlebot I signed it!

**Comment by comment-bot-dev** [comment]
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by Jberlinsky** [comment]
@LukaszCzarnotaSabre Looks like this change is causing test failures:

```
Step #48 - "converge node-pool-local":        Error: Invalid index
Step #48 - "converge node-pool-local":        
Step #48 - "converge node-pool-local":          on ../../../modules/beta-public-cluster/cluster.tf line 423, in resource "google_container_node_pool" "pools":
Step #48 - "converge node-pool-local":         423:       for_each = tobool((lookup(each.value, "sandbox_enabled", ((local.cluster_sandbox_enabled[0] == "gvisor") ? true : false)))) ? ["gvisor"] : []
Step #48 - "converge node-pool-local":            ├────────────────
Step #48 - "converge node-pool-local":            │ local.cluster_sandbox_enabled is empty list of string
Step #48 - "converge node-pool-local":        
Step #48 - "converge node-pool-local":        The given key does not identify an element in this collection value: the
Step #48 - "converge node-pool-local":        collection has no elements.
Step #9 - "converge safer-cluster-local":        
Step #9 - "converge safer-cluster-local":        Error: Invalid index
Step #9 - "converge safer-cluster-local":        
Step #9 - "converge safer-cluster-local":          on ../../../modules/beta-private-cluster/cluster.tf line 442, in resource "google_container_node_pool" "pools":
Step #9 - "converge safer-cluster-local":         442:       for_each = tobool((lookup(each.value, "sandbox_enabled", ((local.cluster_sandbox_enabled[0] == "gvisor") ? true : false)))) ? ["gvisor"] : []
Step #9 - "converge safer-cluster-local":            ├────────────────
Step #9 - "converge safer-cluster-local":            │ local.cluster_sandbox_enabled is empty list of string
Step #9 - "converge safer-cluster-local":        
Step #9 - "converge safer-cluster-local":        The given key does not identify an element in this collection value: the
Step #9 - "converge safer-cluster-local":        collection has no elements.
Step #48 - "converge node-pool-local":        
Step #48 - "converge node-pool-local":        Error: Invalid index
Step #48 - "converge node-pool-local":        
Step #48 - "converge node-pool-local":          on ../../../modules/beta-public-cluster/cluster.tf line 423, in resource "google_container_node_pool" "pools":
Step #48 - "converge node-pool-local":         423:       for_each = tobool((lookup(each.value, "sandbox_enabled", ((local.cluster_sandbox_enabled[0] == "gvisor") ? true : false)))) ? ["gvisor"] : []
Step #48 - "converge node-pool-local":            ├────────────────
Step #48 - "converge node-pool-local":            │ local.cluster_sandbox_enabled is empty list of string
Step #48 - "converge node-pool-local":        
Step #48 - "converge node-pool-local":        The given key does not identify an element in this collection value: the
Step #48 - "converge node-pool-local":        collection has no elements.
Step #30 - "converge stub-domains-local":        google_compute_network.main: Creating...
Step #9 - "converge safer-cluster-local": >>>>>> Running the command `terraform apply -auto-approve -lock=true -lock-timeout=0s -input=false -no-color -parallelism=10 -refresh=true  ` failed due to a non-zero exit code of 1.
Step #9 - "converge safer-cluster-local": >>>>>> ------Exception-------
Step #9 - "converge safer-cluster-local": >>>>>> Class: Kitchen::ActionFailed
Step #9 - "converge safer-cluster-local": >>>>>> Message: 1 actions failed.
Step #9 - "converge safer-cluster-local": >>>>>>     Converge failed on instance <safer-cluster-local>.  Please see .kitchen/logs/safer-cluster-local.log for more details
Step #9 - "converge safer-cluster-local": >>>>>> ----------------------
Step #9 - "converge safer-cluster-local": >>>>>> Please see .kitchen/logs/kitchen.log for more details
Step #9 - "converge safer-cluster-local": >>>>>> Also try running `kitchen diagnose --all` for configuration
Step #48 - "converge node-pool-local":        
Step #48 - "converge node-pool-local":        Error: Invalid index
Step #48 - "converge node-pool-local":        
Step #48 - "converge node-pool-local":          on ../../../modules/beta-public-cluster/cluster.tf line 423, in resource "google_container_node_pool" "pools":
Step #48 - "converge node-pool-local":         423:       for_each = tobool((lookup(each.value, "sandbox_enabled", ((local.cluster_sandbox_enabled[0] == "gvisor") ? true : false)))) ? ["gvisor"] : []
Step #48 - "converge node-pool-local":            ├────────────────
Step #48 - "converge node-pool-local":            │ local.cluster_sandbox_enabled is empty list of string
Step #48 - "converge node-pool-local":        
```

Could you please update tests to make sure they're compliant with this change?

File: autogen/main/cluster.tf.tmpl [hunk 1]
```diff
@@ -577,9 +577,9 @@ resource "google_container_node_pool" "pools" {
     }
     {% if beta_cluster %}
     dynamic "sandbox_config" {
-      for_each = local.cluster_sandbox_enabled
-
+      for_each = tobool((lookup(each.value, "sandbox_enabled", ((local.cluster_sandbox_enabled[0] == "gvisor") ? true : false)))) ? ["gvisor"] : []
```
Comment (bharathkkb) [inline autogen/main/cluster.tf.tmpl:580]: We could use the `sandbox_enabled` variable as the default (as opposed to using `local.cluster_sandbox_enabled[0]`) which would simplify this logic.

**Review by bharathkkb — CHANGES_REQUESTED** [review]
Thanks for the PR! It would also be great to add this to [one of our tests](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/examples/node_pool_update_variant/main.tf) to make sure its WAI

**Comment by bharathkkb** [comment]
@googlebot I consent.

**Review by morgante — APPROVED** [review]

**Pull request merged**